### ISSUE TITLE
Prompt for not simple requests on Web

### ISSUE THREAD

**Issue by AlexV525** [cfug/dio#2201]
### Request Statement

People usually wonder why their requests are being treated as CORS. Various of configuration could lead to the result before people can realize, such as `connectTimeout` and `sendTimeout` across platforms.

### Solution Brainstorm

We might find relevant cases to add logs about "You're making a complex Web request which might lead to CORS" or add more messages when specific exceptions were thrown on the Web platform.

**Comment by kuhnroyal** [comment]
From what I remember, there are no useful exceptions in a lot of cases. Wonder if we could add some sanity checks for common problems and provide suggestions.

**Comment by AlexV525** [comment]
> Wonder if we could add some sanity checks for common problems and provide suggestions.

Exactly, and that would be the thing we'll feel great because the community would love such sweet words
